hkmc2/shared/src/test/mlscript-compile/apps/parsing/Rules.mls

```diff
@@ -260,8 +246,20 @@ val typeRule = rule of
       keyword(Keywords._rightRound) of // either an identifier or nothing
         end of None
         ident of process: someFirst, name: "the type constructor's name"
-  // Infix rules
-  typeExpr of process: pipeInto, choices: typeInfixRule
+  // "infix" rule starting with <type>
+  typeExpr of
+    process: pipeInto
```

Comment:
`pipeInto` should probably be a default value of the parameter. You can implement this logic easily since it's a named parameter.

Reply:
Oh, I see.